Nos/Service/DeepLinkService.swift

```diff
@@ -1,26 +1,64 @@
 import Foundation
 import Logger
+import Dependencies
 
 enum DeepLinkService {
     
-    static let nosURLScheme = "nos"
+    /// Returns the URL scheme for Nos, which varies by build (dev, staging, production). 
+    static var nosURLScheme: String? = { 
+        if let urlTypes = Bundle.main.infoDictionary?["CFBundleURLTypes"] as? [[String: Any]] {
+            for urlTypeDictionary in urlTypes {
+                guard let urlSchemes = urlTypeDictionary["CFBundleURLSchemes"] as? [String] else { continue }
+                guard let externalURLScheme = urlSchemes.first else { continue }
+                return externalURLScheme
+            }
+        }
+        
+        return nil
+    }()
     
     @MainActor static func handle(_ url: URL, router: Router) {
+        @Dependency(\.persistenceController) var persistenceController
+        @Dependency(\.router) var router
         Log.info("handling link \(url.absoluteString)")
         
         let components = URLComponents(url: url, resolvingAgainstBaseURL: true)
         
-        guard let components else {
+        guard let components, let nosURLScheme, components.scheme == nosURLScheme else {
             return
         }
         
-        if components.scheme == nosURLScheme, components.host == "note", components.path == "/new" {
+        if components.host == "note", components.path == "/new" {
             let noteContents = components
                 .queryItems?
                 .first(where: { $0.name == "contents" })?
                 .value
             
             router.showNewNoteView(contents: noteContents)
+        } else {
+            /// Check for links like nos:nevent123174
+            let firstPathComponent = components.path
+            let unformattedRegex = /(?:nostr:)?(?<entity>((npub1|note1|nprofile1|nevent1)[a-zA-Z0-9]{58,255}))/
```

Comment:
Can we move this regex to a new shared place? Even though the regex could be not 100% the same and could not be shared, we are repeating patterns for some things and I found myself sometimes having to fix a problem with one regex and having to search for the other ones to see if the same error could occur there. Just a centralized place where we put all regexes could help to identify potential issues.

Reply:
I want to add support for `nostr:` too. I considered refactoring NoteParser and this code to share a regex, but I didn't because this was just a few minutes of work that I didn't want to throw away because it seemed really useful. I didn't want to (probably) quadruple the time I spent on it to do a refactor. I tried to pull out the shared part of the regex into a constant, but you can't interpolate a regex literal inside another regex literal, so I would have to switch to `NSRegularExpression` or something, which would require more code refactoring... and that would only be for the regex itself. To share the code that extracts the hex from bech32 would require some more.

Given that this is the second place we are using the regex I'm ok with duplicating it. I like the rule of waiting to refactor until you've copied code in _three_ places rather than two.

Reply:
Fair enough. Let's keep the eyes open for the next time we need to add or modify a regex to include that refactor in the scope of the ticket.